### ISSUE TITLE
[BUG] Render stream is blank for non "localhost" IP

### ISSUE THREAD

**Issue by tkbala** [Unity-Technologies/UnityRenderStreaming#502]
**Describe the bug**
I am trying to run the web browser input scene and get it streamed to the video player sample web client. It is working with localhost, but not when the web browser client is opened on a different computer. The webpage loads up, but the stream is just black.

**To Reproduce**
1. Start the webserver using the specified port 
![image](https://user-images.githubusercontent.com/4410442/122457746-d65c0980-cf63-11eb-869f-eb7b15884a75.png)

2. In the webbrowserInput scene in Unity, set the port in the render streaming game object to the desired port and start the program
![image](https://user-images.githubusercontent.com/4410442/122457869-fd1a4000-cf63-11eb-957a-e754bfed02fd.png)1

3. On the remote computer open the video player sample and click play
![image](https://user-images.githubusercontent.com/4410442/122458129-51bdbb00-cf64-11eb-83b2-f86915f23a63.png)

**Expected behavior**
A working render stream video
.

**Environment (please complete the following information):**
 - OS: Both Computers use Windows 
 - Browser : Both computers use chrome browser
 - Unity Version 2019.4.26f1
 - Package version 3.0.1

**Comment by cvachha** [comment]
I am having the same issue. Is there any fix?

**Comment by karasusan** [comment]
@cvachha 
@tkbala
Is the signaling process completed?
Can you disable the firewall to check ports opening or not?

**Comment by cvachha** [comment]
I have tried disabling the firewall (there does not seem to be a difference when I turn it on or off) and I have noticed that when I click play on the web client videoplayer on the remote machine, the Unity console prints 10 new debug.log statements and indicates that the signaling process is connected “Signaling: HTTP connected...”
![image](https://user-images.githubusercontent.com/9502341/133987724-047bd326-6413-492a-89c8-1ecf2fa4440c.png)

However, on the remote machine, the render stream is still empty and the input buttons (“Light off”) do not work. If I run the web client on my local machine, the similar debug.log statements show up in the unity console but the render stream shows up and input works.

I am running the webserver on the same port number of the signaling URL and on the remote machine, The remote machine is on a different network (using a mobile hotspot) and port forwarding into the same port into the network I am running the render streaming server in.

**Comment by karasusan** [comment]
@tkbala @cvachha
Hi, could you answer questions above?

**Comment by Haohua-Lyu** [comment]
Hi, I encountered the same issue and possibly identified why this happens. So when using Http signaling, I found that the unity end sent 3-4 candidates, each with a different IP address (local, public, etc). In our case, the first candidate Unity sends out is always the local IP, and somehow the web browser is "hooked" to the first candidate, which includes a local IP address. Thus the streaming only works when the Unity app and the web client are in the same LAN, but not across WAN. I was able to circumvent this by switching to ws signaling.

**Comment by cvachha** [comment]
The PCs are not in the same network and the issues happen with different ports. This issue is present in the samples that include the videoplayer, where it is blank in the browsers and this seems to happen when using HTTP signaling.

**Comment by karasusan** [comment]
@Haohua-Lyu @cvachha
Sure, we will check the signaling process across WAN to reproduce it.

**Comment by karasusan** [comment]
memo: URS-369

**Comment by scode-cmd** [comment]
This happens even with websocket signaling. Same issue, signaling and streaming works just fine when both client and server are connected to the same network, but render stream is blnk when they are on different network (but signaling works fine)

**Comment by karasusan** [comment]
@scode-cmd
Sure. We will check them.

**Comment by karasusan** [comment]
@tkbala @scode-cmd

I am testing signaling and streaming across WAN and I found the error when using websocket. It could be same this issue.
Is there a console log on your browser?

**Comment by karasusan** [comment]
@tkbala @scode-cmd @Haohua-Lyu @cvachha
Hi, have you solved the issue using TURN/STUN server?

**Comment by Haohua-Lyu** [comment]
Hi, we didn't encounter this bug after shifting to ws signaling, but I do remember we had TURN/STUN servers in place when using HTTP signaling. Once we changed to ws signaling (with no changes on TURN/STUN), the streaming started to work. Hope this helps!

**Comment by eaberman** [comment]
I am having this issue as well, even using web sockets and a working TURN server.

I have deployed the Unity Render Streaming Web App on an EC2 Instance on AWS, as well as a TURN server on an EC2 Instance on AWS. I verified that the TURN server is working using Trickle ICE.

The web stream ALWAYS works when I am on the same network as the browser I have my web app on, but NEVER when I am not on the same network. My Unity console and the requests I see on the web browser using developer tools show that SDP requests are exchanged, and ICE candidates are sent and received (including a relay candidate on TURN server), yet the stream remains dark on the Unity Video Receiver page in the browser.

Can you please advise me on any updates or ways to fix this issue?

**Comment by eaberman** [comment]
Hi @karasusan! Thank you for responding to me.

I believe I was able to replicated the problem without Unity. I access the Web App via Google Chrome, and have built the Unity app with the Render Streaming camera w/signaling manager and broadcast components to an iOS device, and tried to connect. I am able to replicate the issue.

Using Chrome developer tools I have gained more clarity on the problem:
- I can observe the Web App sending ICE Server candidates of type `host`, `srflx`, and `relay` to the Unity iOS app
- I can observe the Unity iOS app sending two candidates of type `host`, but never of any other type

I would expect the Unity iOS app to also send candidates of type `srflx` and `relay` because I had specified a STUN and TURN server. I verified the TURN server is working on TrickerICE, and the Web App is pointed at the same TURN server, and it is successful in gathering a relay candidate.

For some reason, I do not see the Unity iOS app sending any non-`host` ICE Servers. Do you have any insight as to what could be causing this issue?

<img width="464" alt="UnityServers" src="https://user-images.githubusercontent.com/38052620/231195061-c8553395-4ec9-43cc-8bcf-63be809b369d.png">

**Comment by karasusan** [comment]
@eaberman
This is an interesting result! 
I'll checking this later.

**Comment by kannan-xiao4** [comment]
memo: WRS-468

**Comment by eaberman** [comment]
Thank you @karasusan! I appreciate it!

**Comment by eaberman** [comment]
Hi @karasusan, I wanted to add an update to this. I believe that I fixed the issue I was having.

I was specifying my TURN and STUN servers in Render Streaming Project Settings, and marked in the Signaling Manager object attached to my camera to use the default settings in project settings.

But, in the `_Run` function in `Signaling Manager`, this line of code returned an empty array:

`RTCIceServer[] iceServers = settings.iceServers.OfType<RTCIceServer>().ToArray();`

I found that the `settings.iceServers` array in my project was actually of type `IceServer` instead of `RTCIceServer`, so I fixed it by adding code to convert each `IceServer` to an `RTCIceServer` and construct an `RTCIceServer[] iceServers` array from them, and then it appeared to work!

Also, I replaced the line `_conf = conf.GetValueOrDefault(new RTCConfiguration { iceServers = iceServers });` in the `_Run` function with simply `_conf = new RTCConfiguration { iceServers = iceServers };` to ensure the new array was being used.

This seemed to fix my problem. Thank you for responding to my comment on this issue!
